### ISSUE TITLE
Fix remaining Asterisk errors

### ISSUE THREAD

**Issue by TECH7Fox** [TECH7Fox/Asterisk-add-on#23]
There are still some errors when starting Asterisk. Maybe we can fix some.

```
[Jan  4 19:19:15] WARNING[301]: loader.c:2381 load_modules: Some non-required modules failed to load.
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: res_timing_dahdi declined to load.
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: Failed to resolve dependencies for res_stir_shaken
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: res_stir_shaken declined to load.
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: res_pjsip_transport_websocket declined to load.
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: cel_sqlite3_custom declined to load.
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: cdr_sqlite3_custom declined to load.
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: Failed to resolve dependencies for res_http_media_cache
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: res_http_media_cache declined to load.
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: Failed to resolve dependencies for res_pjsip_stir_shaken
[Jan  4 19:19:15] ERROR[301]: loader.c:2396 load_modules: res_pjsip_stir_shaken declined to load.
[Jan  4 19:19:15] WARNING[385]: chan_sip.c:35461 deprecation_notice: chan_sip has no official maintainer and is deprecated.  Migration to
[Jan  4 19:19:15] WARNING[385]: chan_sip.c:35462 deprecation_notice: chan_pjsip is recommended.  See guides at the Asterisk Wiki:
[Jan  4 19:19:15] WARNING[385]: chan_sip.c:35463 deprecation_notice: https://wiki.asterisk.org/wiki/display/AST/Migrating+from+chan_sip+to+res_pjsip
[Jan  4 19:19:15] WARNING[385]: chan_sip.c:35464 deprecation_notice: https://wiki.asterisk.org/wiki/display/AST/Configuring+res_pjsip
```

**Comment by TECH7Fox** [comment]
ERROR[387]: codec_dahdi.c:813 find_transcoders: Failed to open /dev/dahdi/transcode: No such file or directory

**Comment by TimpP-Spb** [comment]
Hi, I have same error when I start Astersk

**Comment by felipecrs** [comment]
Most of them will probably be fixed with #17. But I have some others not mentioned here:

```log
[Feb 18 01:29:15] ERROR[682][C-00000002]: netsock2.c:303 ast_sockaddr_resolve: getaddrinfo("svaiup7tq184.invalid", "(null)", ...): Name does not resolve
[Feb 18 01:29:15] WARNING[682][C-00000002]: chan_sip.c:16938 __set_address_from_contact: Invalid host name in Contact: (can't resolve in DNS) : 'svaiup7tq184.invalid'
[Feb 18 01:29:15] ERROR[682][C-00000002]: netsock2.c:96 ast_sockaddr_stringify_fmt: getnameinfo(): Unrecognized address family or invalid length
[Feb 18 01:29:45] WARNING[675]: res_http_websocket.c:559 ws_safe_read: Web socket closed abruptly
[Feb 18 01:29:45] WARNING[683]: res_http_websocket.c:559 ws_safe_read: Web socket closed abruptly
[Feb 18 01:29:47] ERROR[687]: iostream.c:552 ast_iostream_close: SSL_shutdown() failed: error:00000006:lib(0):func(0):EVP lib, Unknown
```

**Comment by nanosonde** [comment]
It seems that this one is not so easy to fix:

```
[Mar  7 14:52:58] ERROR[558]: res_pjsip/config_system.c:267 system_create_resolver_and_set_nameservers: There are no local system nameservers configured, resorting to system resolution
[Mar  7 14:52:58] ERROR[558]: res_pjsip/config_system.c:267 system_create_resolver_and_set_nameservers: There are no local system nameservers configured, resorting to system resolution
```

Seems to be an issue if the used libc does not provide a certain functionality.
See here: https://github.com/openwrt/telephony/issues/516
What does Alpine use?

**Comment by nanosonde** [comment]
Again concerning the nameserver error message.
The HA addon docker base images really musl as C library.
So asterisk is affected by [this issue](https://github.com/openwrt/telephony/issues/516).

See here: https://github.com/hassio-addons/addon-base/blob/main/base/Dockerfile#L32

**Comment by felipecrs** [comment]
Interesting, is there an alternative to musl?

**Comment by felipecrs** [comment]
Yes, https://github.com/hassio-addons/addon-debian-base.

**Comment by nanosonde** [comment]
So we should consider using a Debian Base image then.

What we should not do under all circumctances is this: glibc-based Alpine with a mixture of two C libraries.
See: https://ariadne.space/2021/08/26/there-is-no-such-thing-as-a-glibc-based-alpine-image/

**Comment by felipecrs** [comment]
Alright, I'm fine either case. We will just have to install Python in the Debian image.

**Comment by nanosonde** [comment]
Gives us more control on the features we really need in the addon as most modules are probably not needed for the use-case described here.

**Comment by felipecrs** [comment]
Interesting... how do we put this in this repo? Copying and pasting? Using git submodule?

**Comment by nanosonde** [comment]
I would suggest to copy and paste the relevant stuff from the Dockerfile and copy the the build_asterisk.sh script.
Looking at the build script, it does the standard stuff.
Furthermore it is also suitbale for cross-compilation (see BUILD_NATIVE) and enables the OPUS codec.

**Comment by nanosonde** [comment]
As this is standard stuff I would just recreate it.
The commands needed to compile asterisk with the various options are also documented on the asterisk documentation page itself.  So there is nothing really new invented here.

**Comment by nanosonde** [comment]
@felipecrs 
Lets use this build script.
Its published under MIT license.

https://github.com/andrius/asterisk/blob/master/debian/18-current/build-asterisk.sh

looks familiar, hm? 😉

**Comment by felipecrs** [comment]
@nanosonde that's awesome!! And yes. Very familiar lol

**Comment by felipecrs** [comment]
[Mar 17 17:23:14] WARNING[476]: res_phoneprov.c:1233 get_defaults: Unable to find a valid server address or name.

[Mar 17 17:23:14] NOTICE[476]: res_smdi.c:1424 load_module: No SMDI interfaces are available to listen on, not starting SMDI listener.

**Comment by nanosonde** [comment]
Damn. Skipped through.
Could be disabled by default.

**Comment by felipecrs** [comment]
```
[Mar 17 17:27:37] ERROR[476]: ari/config.c:312 process_config: No configured users for ARI
```

Also.

**Comment by CedricAndry** [comment]
Hello,
I still have these errors and warning when starting Asterisk 20.5.0 in Docker container based on Alpine (https://registry.hub.docker.com/r/andrius/asterisk), hosted on a Synology NAS (network HOST) :

ERROR[1]: loader.c:2524 load_modules: Failed to resolve dependencies for res_stir_shaken
ERROR[105]: netsock2.c:303 ast_sockaddr_resolve: getaddrinfo("asterisk", "(null)", ...): Name does not resolve
ERROR[105]: ari/config.c:312 process_config: No configured users for ARI
ERROR[106]: res_pjsip/config_system.c:267 system_create_resolver_and_set_nameservers: There are no local system nameservers 

WARNING[105]: res_phoneprov.c:1249 get_defaults: Unable to find a valid server address or name.

Any idea if it will be solved?
Does these errors prevent Asterisk from working properly?

**Comment by felipecrs** [comment]
@CedricAndry I am confused. Are you reporting issues for another image?

And as far as I can tell they do not prevent Asterisk from working.

**Comment by CedricAndry** [comment]
@felipecrs , yes, I am really sorry, I made a mistake.

Thanks for your answer by the way!

Maybe can I afford another question : it is normal that their is no socket opened waiting on the default port (5060) when I launched a command netstat -a | grep 5060 ?

**Comment by felipecrs** [comment]
I am not sure, but it could be. This add-on ships pre-configuration for Asterisk, which you can see at:

- https://github.com/TECH7Fox/asterisk-hass-addons/tree/main/asterisk/rootfs/etc/asterisk
- https://github.com/TECH7Fox/asterisk-hass-addons/tree/main/asterisk/rootfs/usr/share/tempio

But you can also use this add-on with docker: https://github.com/TECH7Fox/asterisk-hass-addons/blob/main/asterisk/CHANGELOG.md#400

**Comment by CedricAndry** [comment]
Ok, thanks! I will try it!

**Comment by WalidDevRes** [comment]
Hello, i'm trying to install astrisk but i have the same error how you resolv it ? 

```
Dec 14 15:29:36 tp asterisk[61765]: [Dec 14 15:29:36] ERROR[61765]: loader.c:2524 load_modules: res_timing_dahdi declined to load.
Dec 14 15:29:36 tp asterisk[61765]: [Dec 14 15:29:36] ERROR[61765]: loader.c:2524 load_modules: cdr_sqlite3_custom declined to load.
Dec 14 15:29:36 tp asterisk[61765]: [Dec 14 15:29:36] ERROR[61765]: loader.c:2524 load_modules: cel_tds declined to load.
Dec 14 15:29:36 tp asterisk[61765]: [Dec 14 15:29:36] ERROR[61765]: loader.c:2524 load_modules: cdr_tds declined to load.
Dec 14 15:29:36 tp asterisk[61765]: [Dec 14 15:29:36] ERROR[61765]: loader.c:2524 load_modules: cel_sqlite3_custom declined to load.
Dec 14 15:29:36 tp asterisk[61765]: [Dec 14 15:29:36] ERROR[61765]: loader.c:2524 load_modules: cdr_pgsql declined to load.
```

**Comment by felipecrs** [comment]
These errors are not blocking. Isn't it working for you?

**Comment by WalidDevRes** [comment]
Yeah it bloking the starting of the service
systemctl restart asterisk
```
root@tp:/usr/src/asterisk-20.5.0# systemctl restart asterisk
Job for asterisk.service failed because a timeout was exceeded.
See "systemctl status asterisk.service" and "journalctl -xe" for details.
root@tp:/usr/src/asterisk-20.5.0#
``` 

journalctl -xe

```
root@tp:/usr/src/asterisk-20.5.0# journalctl -xe
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: chan_dahdi.c:19587 process_dahdi: Ignoring any changes to 'userbase' (on reload) at line 23.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: chan_dahdi.c:19587 process_dahdi: Ignoring any changes to 'vmsecret' (on reload) at line 31.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: chan_dahdi.c:19587 process_dahdi: Ignoring any changes to 'hassip' (on reload) at line 35.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: chan_dahdi.c:19587 process_dahdi: Ignoring any changes to 'hasiax' (on reload) at line 39.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: chan_dahdi.c:19587 process_dahdi: Ignoring any changes to 'hasmanager' (on reload) at line 47.
Dec 14 15:38:43 tp asterisk[62291]: Cannot connect to server socket err = No such file or directory
Dec 14 15:38:43 tp asterisk[62291]: Cannot connect to server request channel
Dec 14 15:38:43 tp asterisk[62291]: jack server is not running or cannot be started
Dec 14 15:38:43 tp asterisk[62291]: JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Dec 14 15:38:43 tp asterisk[62291]: JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Dec 14 15:38:43 tp asterisk[62291]: Cannot connect to server socket err = No such file or directory
Dec 14 15:38:43 tp asterisk[62291]: Cannot connect to server request channel
Dec 14 15:38:43 tp asterisk[62291]: jack server is not running or cannot be started
Dec 14 15:38:43 tp asterisk[62291]: JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Dec 14 15:38:43 tp asterisk[62291]: JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib pcm_a52.c:823:(_snd_pcm_a52_open) a52 is only for playback
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib setup.c:547:(add_elem) Cannot obtain info for CTL elem (MIXER,'IEC958 Playback Default',0,0,0): No such file or directory
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
Dec 14 15:38:43 tp asterisk[62291]: ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
Dec 14 15:38:43 tp asterisk[62291]: Cannot connect to server socket err = No such file or directory
Dec 14 15:38:43 tp asterisk[62291]: Cannot connect to server request channel
Dec 14 15:38:43 tp asterisk[62291]: jack server is not running or cannot be started
Dec 14 15:38:43 tp asterisk[62291]: JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Dec 14 15:38:43 tp asterisk[62291]: JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: ari/config.c:312 process_config: No configured users for ARI
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: confbridge/conf_config_parser.c:2386 verify_default_profiles: Adding default_menu menu to app_confbridge
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: cel_custom.c:92 load_config: No mappings found in cel_custom.conf. Not logging CEL to custom CSVs.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: cel_tds.c:447 tds_load_module: cel_tds has no global category, nothing to configure.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] WARNING[62291]: cel_tds.c:552 load_module: cel_tds module had config problems; declining load
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] WARNING[62291]: cel_pgsql.c:463 process_my_load_module: CEL pgsql config file missing global section.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] NOTICE[62291]: cdr_pgsql.c:541 config_module: cdr_pgsql configuration contains no global section, skipping module load.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: codec_dahdi.c:813 find_transcoders: Failed to open /dev/dahdi/transcode: No such file or directory
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] WARNING[62291]: loader.c:2409 load_modules: Some non-required modules failed to load.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] WARNING[62291]: loader.c:2503 load_modules: Module 'res_adsi' has been loaded but may be removed in a future release.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] WARNING[62291]: loader.c:2503 load_modules: Module 'app_getcpeid' has been loaded but may be removed in a future release.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] WARNING[62291]: loader.c:2503 load_modules: Module 'app_adsiprog' has been loaded but may be removed in a future release.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: loader.c:2524 load_modules: res_timing_dahdi declined to load.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: loader.c:2524 load_modules: cdr_sqlite3_custom declined to load.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: loader.c:2524 load_modules: cel_tds declined to load.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: loader.c:2524 load_modules: cdr_tds declined to load.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: loader.c:2524 load_modules: cel_sqlite3_custom declined to load.
Dec 14 15:38:43 tp asterisk[62291]: [Dec 14 15:38:43] ERROR[62291]: loader.c:2524 load_modules: cdr_pgsql declined to load.
Dec 14 15:38:43 tp asterisk[62291]: Asterisk Ready.
lines 1306-1352/1352 (END)
```

**Comment by felipecrs** [comment]
systemctl? Are we still talking about the Asterisk add-on for Home Assistant?

**Comment by WalidDevRes** [comment]
i'm doing a asterisk server for my home assitance it just for learn about voip

**Comment by felipecrs** [comment]
If you are not using the add-on then you are posting your question in the wrong place. And sorry, I don't have any idea to help with your problem.

**Comment by WalidDevRes** [comment]
but what is the add-on ?

**Comment by WalidDevRes** [comment]
Thanks you for your help
